### PR TITLE
[PLNSRVCE-102] Setup GitOps repo for managing Triggers - Gitops changes + clean up

### PR DESCRIPTION
**Pull request by bnallapeta**
Local Run:

```
[bnr@bnr ckcp]$ ./gitops.sh -a tekton-pipeline -a pipelines -a triggers-crds -a triggers-interceptors -a triggers-controller
[openshift-gitops]
  - OpenShift-GitOps: OK
  - ArgoCD dashboard: .....................................................................................................................................OK
  - ArgoCD URL: https://openshift-gitops-server-openshift-gitops.apps.ci-ln-82yg4k2-72292.origin-ci-int-gce.dev.rhcloud.com
  - ArgoCD Login: OK

[ckcp]
  - ckcp application: OK
  - KCP pod: OK
  - External IP: OK
  - KCP api-server: OK
  - Register KCP secret to host cluster: ..OK
  - Cluster registration: OK
  - Workspace: workspace.tenancy.kcp.dev/demo created
OK
  - ArgoCD KCP registration: OK

[tekton-pipeline]
  - tekton-pipeline application: OK

[pipelines]
  - pipelines application: OK

[triggers-crds]
  - triggers-crds application: OK

[triggers-interceptors]
..  - triggers-interceptors application: OK

[triggers-controller]
..  - triggers-controller application: OK
```

```
bnr@bnr ckcp]$ ./test.sh pipelines triggers
.
.
.
+ KUBECONFIG=/home/bnr/workspace/pipelines-service/ckcp/work/kubeconfig/admin.kubeconfig
+ kubectl get taskruns,pipelineruns
NAME                                              SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
taskrun.tekton.dev/custom-env-8dbj4               True        Succeeded   83s         74s
taskrun.tekton.dev/github-run-l5wcz               True        Succeeded   22s         13s
taskrun.tekton.dev/test-pipelinerun-gsjjd-task1   True        Succeeded   81s         71s

NAME                                            SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
pipelinerun.tekton.dev/test-pipelinerun-gsjjd   True        Succeeded   82s         72s
```

### CODE REVIEW

File: ckcp/gitops.sh [hunk 1]
```diff
@@ -318,7 +312,78 @@ install_triggers(){
...
+
+install_triggers_interceptors(){
+  APP="triggers-interceptors"
+  export KUBECONFIG="$KUBECONFIG_CLUSTER"
+
+  #############################################################################
+  # Create kcp-kubeconfig secrets for event listener and interceptors so that they can talk to KCP
+  #############################################################################
+  if ! KUBECONFIG=$KUBECONFIG_KCP oc get secret kcp-kubeconfig >/dev/null 2>&1; then
+    echo -n "."
```
Comment (Michkov) [inline ckcp/gitops.sh:341]: makes dot on wrong place -> `..  - triggers-interceptors application: OK`. I would just remove it.

File: ckcp/gitops.sh [hunk 2]
```diff
@@ -318,7 +312,78 @@ install_triggers(){
...
+    KUBECONFIG=$KUBECONFIG_KCP oc create secret generic kcp-kubeconfig --from-file "$KUBECONFIG_KCP" >/dev/null
+  fi
+  if ! KUBECONFIG=$KUBECONFIG_KCP oc get secret kcp-kubeconfig -n tekton-pipelines >/dev/null 2>&1; then
+    echo -n "."
```
Comment (Michkov) [inline ckcp/gitops.sh:345]: makes dot on wrong place -> `..  - triggers-interceptors application: OK`. I would just remove it.

File: ckcp/gitops.sh [hunk 3]
```diff
@@ -318,7 +312,78 @@ install_triggers(){
...
+
+install_triggers_controller(){
+  APP="triggers-controller"
+  export KUBECONFIG="$KUBECONFIG_CLUSTER"
+
+  #############################################################################
+  # Create kcp-kubeconfig secret for triggers controller
+  #############################################################################
+  if ! oc get namespace triggers >/dev/null 2>&1; then
+    echo -n "."
```
Comment (Michkov) [inline ckcp/gitops.sh:368]: makes dot on wrong place -> `..  - triggers-controller application: OK`. I would just remove it.

File: ckcp/gitops.sh [hunk 4]
```diff
@@ -318,7 +312,78 @@ install_triggers(){
...
+    oc create namespace triggers >/dev/null
+  fi
+  if ! oc get secret ckcp-kubeconfig -n triggers >/dev/null 2>&1; then
+    echo -n "."
```
Comment (Michkov) [inline ckcp/gitops.sh:372]: makes dot on wrong place -> `..  - triggers-controller application: OK`. I would just remove it.

File: ckcp/test.sh [hunk 5]
```diff
@@ -38,9 +38,16 @@ else
...
     elif [ $arg == "triggers" ]; then
       echo "Arg triggers passed. Running triggers tests..."
+
       echo "Simulating a Github PR through a curl request which creates a TaskRun (from tektoncd/triggers/examples)"
-      KUBECONFIG=$KUBECONFIG_KCP kubectl apply -f triggers/examples/v1beta1/github/
+      mkdir -p $SCRIPT_DIR/github-eventlistener
+      curl https://raw.githubusercontent.com/tektoncd/triggers/v0.18.0/examples/v1beta1/github/github-eventlistener-interceptor.yaml > github-eventlistener/github-eventlistener-interceptor.yaml
+      curl https://raw.githubusercontent.com/tektoncd/triggers/v0.18.0/examples/v1beta1/github/secret.yaml > github-eventlistener/secret.yaml
+
+      KUBECONFIG=$KUBECONFIG_KCP kubectl apply -f github-eventlistener
```
Comment (Michkov) [inline ckcp/test.sh:47]: You can apply directly
```
KUBECONFIG=$KUBECONFIG_KCP kubectl apply -f https://raw.githubusercontent.com/tektoncd/triggers/v0.18.0/examples/v1beta1/github/github-eventlistener-interceptor.yaml
KUBECONFIG=$KUBECONFIG_KCP kubectl apply -f https://raw.githubusercontent.com/tektoncd/triggers/v0.18.0/examples/v1beta1/github/secret.yaml
```

File: ckcp/gitops.sh (hunk 1 above)
Comment (bnallapeta) [reply ckcp/gitops.sh:341]: Removed the dot everywhere.

File: ckcp/test.sh (hunk 5 above)
Comment (bnallapeta) [reply ckcp/test.sh]: Done.

File: ckcp/gitops.sh [hunk 6]
```diff
@@ -318,7 +312,74 @@ install_triggers(){
   echo
```
Comment (Michkov) [inline ckcp/gitops.sh:312]: is this still needed?

Comment (bnallapeta) [reply ckcp/gitops.sh]: Removed.

**Pull request merged**